pkg/volume/emptydir/empty_dir.go

```diff
@@ -307,37 +308,54 @@ func (ed *emptyDir) setupHugepages(dir string) error {
 	return ed.mounter.Mount("nodev", dir, "hugetlbfs", []string{pageSizeMountOption})
 }
 
-// getPageSizeMountOptionFromPod retrieves pageSize mount option from Pod's resources
-// and validates pageSize options in all containers of given Pod.
-func getPageSizeMountOptionFromPod(pod *v1.Pod) (string, error) {
+// getPageSizeMountOption retrieves pageSize mount option from Pod's resources
+// and medium and validates pageSize options in all containers of given Pod.
+func getPageSizeMountOption(medium v1.StorageMedium, pod *v1.Pod) (string, error) {
 	pageSizeFound := false
 	pageSize := resource.Quantity{}
-	// In some rare cases init containers can also consume Huge pages.
-	containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
-	for _, container := range containers {
+
+	var mediumPageSize resource.Quantity
+	var err error
+	if medium != v1.StorageMediumHugePages {
+		// medium is: Hugepages-<size>
+		mediumPageSize, err = v1helper.HugePageSizeFromMedium(medium)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// In some rare cases init containers can also consume Huge pages
+	for _, container := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
 		// We can take request because limit and requests must match.
 		for requestName := range container.Resources.Requests {
 			if v1helper.IsHugePageResourceName(requestName) {
 				currentPageSize, err := v1helper.HugePageSizeFromResourceName(requestName)
 				if err != nil {
 					return "", err
 				}
-				// PageSize for all volumes in a POD are equal, except for the first one discovered.
-				if pageSizeFound && pageSize.Cmp(currentPageSize) != 0 {
-					return "", fmt.Errorf("multiple pageSizes for huge pages in a single PodSpec")
+				if medium == v1.StorageMediumHugePages { // medium is: Hugepages, size is not specified
+					// PageSize for all volumes in a POD must be equal if medium is "Hugepages"
+					if pageSizeFound && pageSize.Cmp(currentPageSize) != 0 {
+						return "", fmt.Errorf("multiple pageSizes for huge pages in a single PodSpec")
+					}
+
+					pageSizeFound = true
+					pageSize = currentPageSize
+				} else { // medium is: Hugepages-<size>
+					if currentPageSize.Cmp(mediumPageSize) == 0 {
```

Comment:
Ref. comment above, I think this fits better in the pod validation, and not here. For volumes with medium `HugePages-<size>` we already know the page size, so it should be unnecessary to run this double loop.

Reply:
I think it can be done in PodSpec or Pod validation. However, in this case empty_dir code will be left without validation, which some people wouldn't like I guess. WDYT?

Reply:
I'm going to move validation to the validation.go in a separate PR as you've suggested.